### PR TITLE
Multi-Width to support Sketch with multiple wires

### PR DESCRIPTION
**Pull request by paullee0**
Discussion:-
https://forum.freecadweb.org/viewtopic.php?f=23&t=37708&p=321956#p321956

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, please confirm the following:

- [x] Branch rebased on latest master `git pull --rebase upstream master`
- [ ] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)
- [ ] Commit message includes `issue #<id>` or `fixes #<id>` where `<id>` is the [associated MantisBT](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) issue id if one exists

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

### CODE REVIEW

File: src/Mod/Arch/ArchWall.py [hunk 1]
```diff
@@ -944,12 +949,16 @@ def getExtrusionData(self,obj):
...
-                                    overrideWidthHalfen = [i/2 for i in obj.OverrideWidth]
+                                    overrideWidthHalfen = [i/2 for i in widths]
+
                                     w1 = DraftGeomUtils.offsetWire(wire,dvec,False, False, overrideWidthHalfen)
                                     dvec = dvec.negative()
                                     w2 = DraftGeomUtils.offsetWire(wire,dvec,False, False, overrideWidthHalfen)
 
                                 sh = DraftGeomUtils.bind(w1,w2)
+
+                            del widths[0:edgeNum]
```
Comment (yorikvanhavre) [inline src/Mod/Arch/ArchWall.py]: Isn't this deleting the entire widths content? Isn't it better to simply `del widths` or `widths = []`? Might be a bit faster

Comment (paullee0) [reply src/Mod/Arch/ArchWall.py]: Hmm, not exactly, see if i can explain better below:-

The whole list can contain width for every 'segment'/edge of wire in the sketch / wall.

For every 'loop' in 896 
`for i,wire in enumerate(self.basewires):`
... a 'continuous' series of edges / i.e. a wire is processed.

Then, 960 only take out from the list of "widths" the numbers of width figures corresponding to the numbers of edges in the wire.

Comment (paullee0) [reply src/Mod/Arch/ArchWall.py]: see also 898
edgeNum = len(wire.Edges)

then 960
del widths[0:edgeNum]

**Comment by yorikvanhavre** [comment]
Okay! Merging then

**Comment by paullee0** [comment]
Thanks.  I try to see if any of these leads to the recent 0.19_pre Arch Wall creation problem base on Sketch but have no idea at all.

**Pull request merged**